base/dates/periods.jl

```diff
@@ -166,10 +166,55 @@ type CompoundPeriod <: AbstractTime
                     break
                 end
             end
-            return new(reverse!(pc))
+            p = reverse!(pc)
+            n = length(p)
         else
             return new(resize!(p, n))
         end
+        # reduce the amount of mixed positive/negative Periods.
+        if n > 0
+            pc = sizehint!(Period[], n)
+            i = n
+            while i > 0
```

Comment:
Oh I see... you're imagining a case where we might have two disjoint sets of fungible periods, like years and centuries vs. hours and minutes.